src/mediaLibrary/MediaLibraryVideoClient.ts

```diff
@@ -0,0 +1,217 @@
+  ): Promise<VideoPlaybackInfo> {
+    return lastValueFrom(
+      new ObservableMediaLibraryVideoClient(
+        this.#client.observable,
+        this.#httpRequest,
+      ).getPlaybackInfo(assetIdentifier, options),
+    )
+  }
+
+  /**
+   * Extract playback tokens from signed video playback info
+   *
+   * @param playbackInfo - The playback info response
+   * @returns Object containing playback, thumbnail, and storyboard tokens if signed, undefined otherwise
+   */
+  getPlaybackTokens(playbackInfo: VideoPlaybackInfo): VideoPlaybackTokens | undefined {
+    if (isSignedPlaybackInfo(playbackInfo)) {
+      return {
+        playback: playbackInfo.stream.token,
+        thumbnail: playbackInfo.thumbnail.token,
+        storyboard: playbackInfo.storyboard.token,
+        animated: playbackInfo.animated.token,
+      }
+    }
+
+    return undefined
+  }
+}
+
+function parseAssetInstanceId(assetIdentifier: MediaLibraryAssetInstanceIdentifier): {
+  instanceId: string
+  libraryId?: string
+} {
+  if (typeof assetIdentifier === 'string') {
+    // Handle video-prefixed asset instance ID
+    if (assetIdentifier.startsWith('video-')) {
+      return {instanceId: assetIdentifier}
+    }
+    // Assume it's a container ID or plain instance ID
+    return {instanceId: assetIdentifier}
+  }
+
+  if (assetIdentifier && typeof assetIdentifier === 'object' && '_ref' in assetIdentifier) {
+    const ref = assetIdentifier._ref
+    if (typeof ref === 'string') {
+      // Parse GDR to extract library ID and instance ID
+      // Expected format: "media-library:libraryId:instanceId"
+      const gdrParts = ref.split(':')
+
+      // Valid GDR must have exactly 3 parts and start with 'media-library'
+      if (gdrParts.length === 3 && gdrParts[0] === 'media-library') {
+        const [, libraryId, instanceId] = gdrParts
+
+        // Validate that libraryId and instanceId are not empty
+        if (!libraryId || !instanceId) {
+          throw new Error(
+            `Invalid media library reference format: "${ref}". Expected format: "media-library:mlXXXX:instanceId"`,
+          )
+        }
+
+        // Validate that libraryId starts with 'ml' prefix
+        if (!libraryId.startsWith('ml')) {
+          throw new Error(
+            `Invalid library ID in reference: "${ref}". Library ID must start with "ml" prefix`,
+          )
+        }
+
+        return {
+          libraryId,
+          instanceId,
+        }
+      }
+
+      // Invalid GDR format
+      throw new Error(
+        `Invalid media library reference format: "${ref}". Expected format: "media-library:mlXXXX:instanceId"`,
+      )
+    }
+  }
+
+  throw new Error('Invalid asset identifier: must be a string or an object with a _ref property')
+}
+
+function buildVideoPlaybackInfoUrl(instanceId: string, libraryId?: string): string {
+  const effectiveLibraryId = libraryId || 'default'
+  return `/media-libraries/${effectiveLibraryId}/video/${instanceId}/playback-info`
+}
```

Comment:
Not sure if I'm missing something here but as I understand it `getPlaybackInfo` calls this function to determine the playback URL, and the `assetIdentifier` that `getPlaybackInfo` accepts needs to be a GDR or `libraryId` will be undefined here. In the case the library ID will be `'default'`. Should it not be using the `id` passed via `~experimental_resource`?

Reply:
Ah yes, I thought I fixed this. Will update.